.github/workflows/lint-build-commits.yaml

```diff
@@ -13,7 +13,7 @@ jobs:
   build_commits:
     name: Check if build works for every commit
     runs-on: ubuntu-22.04
-    timeout-minutes: 90
+    timeout-minutes: 180
```

Comment:
why the timeout increase?

Reply:
I thought a bit about this. IMO 3 hours is way too long for feedback. But on the other hand this workflow is not marked as required in order to merge changes, so if it's running in the background for a while then it's not the end of the world. I also thought about why this is so slow, I wonder whether we're invalidating some caching that causes each build to take a long time. 

Separately I also figured that we can mitigate this by splitting into smaller PRs with a subset of the commits (for instance if there's a few refactor commits that should be generally OK to improve the codebase even if they don't achieve the full goal of the PR). I recognize that in the linked case, the overall PR is relatively small with a relatively high number of commits... I also wonder whether it may be quicker to review and merge if the PR was split more by the codeowners that are touched by the PR, allowing just one or two reviewers to sign off to merge part of the effort. 

Looking through the example build from the linked PR, it seems like it takes maybe 5m per commit to build userspace then another ~1.5m per commit to build the datapath, so I guess the current settings limit the max number of commits to around 10-12 while still allowing this job to pass. I don't think that's all that crazy either. I don't want to disincentivize plenty of small logical commits if developers are willing to split them up, but I do also wonder whether once we get to 10+ commits if they should be grouped into PRs of around 10 or less just to keep the size down and review throughput up.

In principle I'm not necessarily against this, other than that I just think that waiting this long for CI feedback on a PR is not ideal. (We are trying for about 15m for direct feedback for the basic runs, then I think some of the cloud workflows take perhaps 15m due to 15m provisioning + 15m testing...)